src/View/Antlers/Language/Analyzers/RecursiveParentAnalyzer.php

```diff
@@ -54,6 +54,10 @@ public static function associateRecursiveParent($nodes)
                             if (Str::contains($subNode->runtimeContent, $recursiveContent) && mb_substr_count($subNode->runtimeContent, '*recursive') == 1 && $node->getRootRef() == $subNode->getRootRef()) {
                                 $lastNode = $subNode;
 
+                                if (! ConditionPairAnalyzer::isConditionalStructure($subNode) && $subNode->isClosedBy != null && $subNode->name->name == $node->name->name) {
+                                    break;
+                                }
+
                                 // Ensure we stop searching once we reach the closest nav parent.
                                 if ($subNode->name->name == 'nav') {
```

Comment:
Does this PR allow us to remove this then?

Reply:
I'll test to see if we can safely do that :)